dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/single/SingleItemDataWriter.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.persistence.file.single
+
+import com.datadog.android.core.internal.persistence.DataWriter
+import com.datadog.android.core.internal.persistence.Serializer
+import com.datadog.android.core.internal.persistence.file.FileHandler
+import com.datadog.android.core.internal.persistence.file.FileOrchestrator
+
+internal open class SingleItemDataWriter<T : Any>(
+    internal val fileOrchestrator: FileOrchestrator,
+    internal val serializer: Serializer<T>,
+    internal val handler: FileHandler
+) : DataWriter<T> {
+
+    // region DataWriter
+
+    override fun write(element: T) {
+        consume(element)
+    }
+
+    override fun write(data: List<T>) {
```

Comment:
maybe I will discover it in the next commits, but what is the use-case for this class? like we write only last element and the rest is dropped?